### PR TITLE
Update setup.Ensure*PKI() functions to delete empty certificates

### PR DESCRIPTION
**Pull request by neoaggelos**
### Summary

Remove certificate files if they are empty (such that stale or removed certificates are no longer used).

### Changes

- Reduce some code duplication with helper `ensureFiles()` utility
- remove files if the cert contents are empty

### CODE REVIEW

File: src/k8s/pkg/k8sd/setup/certificates.go [hunk 1]
```diff
@@ -2,120 +2,84 @@ package setup
...
 	"github.com/canonical/k8s/pkg/snap"
 )
 
-func EnsureExtDatastorePKI(snap snap.Snap, certificates *pki.ExternalDatastorePKI) error {
-	toWrite := map[string]string{
-		path.Join(snap.EtcdPKIDir(), "ca.crt"):     certificates.DatastoreCACert,
-		path.Join(snap.EtcdPKIDir(), "client.key"): certificates.DatastoreClientKey,
-		path.Join(snap.EtcdPKIDir(), "client.crt"): certificates.DatastoreClientCert,
+// ensureFile creates fname with the specified contents, mode and owner bits.
+// ensureFile will delete the file if contents is an empty string.
```
Comment (bschimke95) [inline src/k8s/pkg/k8sd/setup/certificates.go:14]: this "side-effect" is never executed, is it? The file path is always set.
As a side-note, I'm not a big fan of this. That behavior is not really obvious from the function signature.

Comment (neoaggelos) [reply src/k8s/pkg/k8sd/setup/certificates.go:14]: we delete the file if `contents` is empty. the `fname` can obviously not be an empty string. or do i not understand your comment?

Comment (bschimke95) [reply src/k8s/pkg/k8sd/setup/certificates.go:14]: nevermind - my comment made no sense...

**Review by bschimke95 — APPROVED** [review]
LGTM

**Pull request merged**